### PR TITLE
Fixed bug with code block.

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**
There's no way to disable the auto closing tags in cheerio per https://github.com/cheeriojs/cheerio/issues/859, so I added a regex to strip it out of the languages that use that generic syntax.

### CODE REVIEW

File: src/code-block.js [hunk 1]
```diff
@@ -15,7 +15,10 @@ const preload = (content, resourcePath) => {
 
   codeBlocks.each((idx, elem) => {
     const $elem = $(elem);
-    const rawContent = $elem.html().toString();
+    let rawContent = $elem.html().toString();
+    if (shared.codeBlockLanguagesWithGenerics.includes($elem.attr('languageType'))) {
+      rawContent = rawContent.replace(/<\/(.*)>/, '');
```
Comment (Blackbaud-BrandonJones) [inline src/code-block.js]: this regex will match the followigin `</some tag> adlskgnaut 1 > 2` and will include that other closing tag.  I think you need something like `/<\/(.*?)>/g` to match all "closing" tags?

File: src/code-block.spec.js [hunk 2]
```diff
@@ -33,4 +33,19 @@ describe('Code Block Plugin', () => {
...
+  it('should remove invalid auto closing tags', () => {
+    const content = new Buffer(`
+      <stache-code-block languageType="csharp">
+        public async Task<string> someMethod()
+        {
+          //Do some stuff
+        }
+      </stache-code-block>
+    `);
+    const path = 'foo.html';
```
Comment (Blackbaud-BrandonJones) [inline src/code-block.spec.js]: Does this strip out the `</stache-code-block>`?

File: src/code-block.js [hunk 3]
```diff
@@ -15,7 +15,10 @@ const preload = (content, resourcePath) => {
 
   codeBlocks.each((idx, elem) => {
     const $elem = $(elem);
-    const rawContent = $elem.html().toString();
+    let rawContent = $elem.html().toString();
+    if (shared.codeBlockLanguagesWithGenerics.includes($elem.attr('languageType'))) {
```
Comment (Blackbaud-BrandonJones) [inline src/code-block.js]: Is there ever a case where these languages would want a closing tag for any reason at all?  i.e Would Java or Csharp ever contain a string template of html?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@blackbaud/skyux-builder-plugin-stache",
-  "version": "1.3.0",
+  "version": "1.3.1",
```
Comment (Blackbaud-BrandonJones) [inline package.json]: I would prefer we create a new "release" branch after these changes get approved an in.  It's easier to backtrack through the commit/PR history if they are separated out, release from feature.

File: src/code-block.spec.js (hunk 2 above)
Comment (blackbaud-brandonhare) [reply src/code-block.spec.js]: Doesn't strip out the code block, since we're looking in the code block. :)

File: src/code-block.js (hunk 3 above)
Comment (blackbaud-brandonhare) [reply src/code-block.js]: Not a concern here.

File: package.json (hunk 4 above)
Comment (blackbaud-brandonhare) [reply package.json]: 🤷‍♀️

Comment (Blackbaud-BrandonJones) [reply package.json]: 🙄  LOL.  It's cleaner and easier to track specific items if a release/version bump branch is separated from the feature change.  When browsing through the commit history we can more easily identify when/where features were added/changed/fixed If they are kept under a different commit heading.  Also when tracking when a specific release went out, having that separate branch is easier to find.

File: src/code-block.js [hunk 5]
```diff
@@ -15,7 +15,10 @@ const preload = (content, resourcePath) => {
...
+      rawContent = rawContent.replace(/<\/(.*?)>/g, '');
```
Comment (Blackbaud-BrandonJones) [inline src/code-block.js]: 👍

File: src/code-block.js (hunk 3 above)
Comment (Blackbaud-BrandonJones) [reply src/code-block.js]: 👍

**Comment by Blackbaud-BrandonJones** [comment]
Just one more final thought.  `Typescript` also supports the idea of using generics.  Any thoughts on being able to extend this functionality to include typescript methods? I don't think we can follow the same pattern of just removing all closing tags in that case.

**Comment by blackbaud-brandonhare** [comment]
Yep. Added typescript and updated to just strip out the end of the string. I think this is good to go. Any input @Blackbaud-SteveBrush ?

**Comment by Blackbaud-BobbyEarl** [comment]
I would consider these definitely edge-cases, so take them as you will, but I'm able to produce some code sample that still exhibit strange behavior.  The first one incorrectly removes the trailing `</p>` and the second one (unrelated to your change) starts adding `=""` to some of the generics.

```
<stache-code-block languageType="typescript">
public async Task<string> someMethod() {
  // Something
}
const myVar2 = `<p></p>`;
</stache-code-block>
```

```
<stache-clode-block languageType="java">
public interface Pair<K, V> {
  public K getKey();
  public V getValue();
}
</stache-code-block>
```

**Pull request closed without merging**